spec/views/report/_form_expression_buttons.html.haml_spec.rb

```diff
@@ -0,0 +1,32 @@
+describe 'report/_form_filter.html.haml' do
+
+  let(:report) { FactoryGirl.create(:miq_report) }
+  let(:expkey) { :record_filter }
+  let(:exp_table) { [["Example AFTER \"value1\"", 1]] }
+  let(:expression) { {"after" => {"field" => "field_example", "value" => "value1"}, :token => 1} }
+  let(:edit) { { :rpt_id => report.id, expkey => {:exp_table => exp_table, :expression => expression}} }
+
+  before :each do
+    assign(:expkey, expkey)
+    assign(:edit, edit)
+  end
+
+  it 'renders form expression buttons for other report and Edit Display Filter button' do
+    render :partial => 'report/form_expression_buttons',
+           :locals  => {:create_label  => _('Create Display Filter'),
+                        :display_label => _('Edit Display Filter')}
+    expect(rendered).to match(/Edit Display Filter/)
+  end
+
+  it 'renders form expression buttons for other report and Create Display Filter button' do
+    exp_table = [["???", 1]]
+    expression = {"???" => "???", :token => 1}
+    edit = {:rpt_id => report.id, expkey => {:exp_table => exp_table, :expression => expression}}
```

Comment:
Rewriting `let` vars is not a good way, you should use a separate `context` for this test.